Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp

```diff
@@ -93,7 +94,16 @@ static WebCore::IntRect screenRectOfContents(WebCore::Element& element)
     auto viewportRect = snappedIntRect(frameView->layoutViewportRect());
     contentsRect.intersect(viewportRect);
 
-    return frameView->contentsToScreen(contentsRect);
+    return frameView->contentsToRootView(contentsRect);
+}
+
+static void convertToRootFrameCoordinates(Element& element, WebCore::FrameIdentifier& frameID, WebCore::IntRect& coordinates)
+{
+    coordinates = rootViewRectOfContents(element);
+    CheckedPtr renderer = element.renderer();
+    if (!renderer)
+        return;
+    frameID = renderer->protectedFrame()->rootFrame().frameID();
```

Comment:
Nit: could be a little more compact
```suggestion
    if (CheckedPtr renderer = element.renderer())
        frameID = renderer->protectedFrame()->rootFrame().frameID();
```